uber/models.py

```diff
@@ -2038,7 +2052,9 @@ class PromoCode(MagModel):
             func.replace(func.replace(func.lower(code), '-', ''), ' ', ''),
             unique=True),
         CheckConstraint(func.trim(code) != '',
-            name='ck_promo_code_non_empty_code'))
+            name='ck_promo_code_non_empty_code'),
+        CheckConstraint(or_(discount != None, uses_allowed != None),
+            name='ck_promo_code_no_unlimited_use_free_badge'))
```

Comment:
I'm fine with it. If an admin really really wants to get around it, they can make a promo code with 500 uses or something, which is actually better and safer than literally unlimited uses.

Generally we don't actually put these things in the database, but I think it's fine.